webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/MeshFileService.scala

```diff
@@ -368,24 +377,38 @@ class MeshFileService @Inject()(config: DataStoreConfig)(implicit ec: ExecutionC
   def readMeshChunkV3(organizationName: String,
                       dataSetName: String,
                       dataLayerName: String,
-                      meshChunkDataRequest: MeshChunkDataRequestV3,
-  ): Fox[(Array[Byte], String)] = {
-    val meshFilePath = dataBaseDir
-      .resolve(organizationName)
-      .resolve(dataSetName)
-      .resolve(dataLayerName)
-      .resolve(meshesDir)
-      .resolve(s"${meshChunkDataRequest.meshFile}.$meshFileExtension")
-
-    executeWithCachedHdf5(meshFilePath, meshFileCache) { cachedMeshFile =>
-      val meshFormat = cachedMeshFile.reader.string().getAttr("/", "mesh_format")
-      val data =
-        cachedMeshFile.reader
-          .uint8()
-          .readArrayBlockWithOffset("neuroglancer", meshChunkDataRequest.byteSize, meshChunkDataRequest.byteOffset)
-      (data, meshFormat)
-    } ?~> "mesh.file.readData.failed"
-  }
+                      meshChunkDataRequests: MeshChunkDataRequestV3List,
+  ): Fox[(Array[Byte], String)] =
+    for {
+      data: List[(Array[Byte], String, Int)] <- Fox.combined(
+        meshChunkDataRequests.requests.zipWithIndex
+          .sortBy(requestAndIndex => requestAndIndex._1.byteOffset)
+          .map(requestAndIndex => {
+            val meshChunkDataRequest = requestAndIndex._1
+            val meshFilePath = dataBaseDir
+              .resolve(organizationName)
+              .resolve(dataSetName)
+              .resolve(dataLayerName)
+              .resolve(meshesDir)
+              .resolve(s"${meshChunkDataRequests.meshFile}.$meshFileExtension")
+
+            executeWithCachedHdf5(meshFilePath, meshFileCache) { cachedMeshFile =>
+              val meshFormat = cachedMeshFile.reader.string().getAttr("/", "mesh_format")
+              val data =
+                cachedMeshFile.reader
+                  .uint8()
+                  .readArrayBlockWithOffset("neuroglancer",
+                                            meshChunkDataRequest.byteSize,
+                                            meshChunkDataRequest.byteOffset)
+              (data, meshFormat, requestAndIndex._2)
+            } ?~> "mesh.file.readData.failed"
+          })
+          .toList)
+      dataSorted = data.sortBy(d => d._3)
+      _ <- Fox.bool2Fox(data.map(d => d._2).toSet.size == 1) // Ensure same encoding for all responses
+      encoding = data.map(d => d._2).head
```

Comment:
```suggestion
      encoding <- data.map(d => d._2).headOption
```
(will create a Fox.failure rather than an exception if something goes wrong)

Reply:
Don't think anything can go wrong because of the statument in the line above, but I guess it is better style